uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -494,13 +509,21 @@ define([
       };
 
       /**
-       * Checks if given status is available to manager
+       * Checks if given status is available to manager. If manager applies leave
+       * on behalf of staff then cancelled is also removed from her list of available statuses.
        *
-       * @param {String} status
-       * @return {Boolean}
+       * @return {Function} that takes status as param and returns a boolean
        */
-      this.isStatusAvailableForManager = function (status) {
-        return !(status.name === 'admin_approved' || status.name === 'waiting_approval');
+      this.isStatusAvailableForManager = function () {
+        var self = this;
+
+        return function (status) {
+          var canRemoveStatus = (status.name === 'admin_approved' || status.name === 'waiting_approval');
+
+          canRemoveStatus = self.directiveOptions.forStaff ? canRemoveStatus || status.name === 'cancelled' : canRemoveStatus;
+
+          return !canRemoveStatus;
+        }
       }
```

Comment:
This is still not fixed. also there are many other places in the file where semicolon is missing(Specially the functions with `this.`). Could you please fix all those also.